src/lexers/javascript-lexer.js

```diff
@@ -57,8 +58,45 @@ export default class JavascriptLexer extends BaseLexer {
 
             that.keys.push(entry)
           }
+        },
+        JSXElement(node) {
+          const element = node.openingElement
+          if (element.name.name === "Trans") {
+            const entry = {}
+            const defaultValue = nodeToString(node, content)
+        
+            element.attributes.forEach(attr => {
+              if (attr.name.name === that.attr) {
+                entry.key = attr.value.value
+              }
+            })
+        
+            if (defaultValue !== '') {
+              entry.defaultValue = defaultValue
+
+              if (!entry.key)
+              entry.key = entry.defaultValue
+            }
+        
+            if (entry.key)
+              that.keys.push(entry)
+          }
+
+          else if (element.name.name === "Interpolate") {
+            const entry = {}
+        
+            element.attributes.forEach(attr => {
+              if (attr.name.name === that.attr) {
+                entry.key = attr.value.value
+              }
+            })
+        
+            if (entry.key)
+              that.keys.push(entry)
+          }
         }
-      }
+      },
+      JSXParserExtension
```

Comment:
We use 'JSXParserExtension' (see below) as a 'base' according to the acorn docs. It's a thing that defines more traps, and how to respond to them.